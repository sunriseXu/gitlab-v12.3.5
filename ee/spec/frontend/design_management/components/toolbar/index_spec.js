import { createLocalVue, shallowMount } from '@vue/test-utils';
import VueRouter from 'vue-router';
import Toolbar from 'ee/design_management/components/toolbar/index.vue';

const localVue = createLocalVue();
localVue.use(VueRouter);
const router = new VueRouter();

const RouterLinkStub = {
  props: {
    to: {
      type: Object,
    },
  },
  render(createElement) {
    return createElement('a', {}, this.$slots.default);
  },
};

describe('Design management toolbar component', () => {
  let wrapper;

  function createComponent(isLoading = false) {
    const updatedAt = new Date();
    updatedAt.setHours(updatedAt.getHours() - 1);

    wrapper = shallowMount(Toolbar, {
      sync: false,
      localVue,
      router,
      propsData: {
        id: '1',
        isLoading,
        name: 'test.jpg',
        updatedAt: updatedAt.toString(),
        updatedBy: {
          name: 'Test Name',
        },
      },
      stubs: {
        'router-link': RouterLinkStub,
      },
    });
  }

  it('renders design and updated data', () => {
    createComponent();

    expect(wrapper.element).toMatchSnapshot();
  });

  it('links back to designs list', () => {
    createComponent();

    const link = wrapper.find('a');

    expect(link.props('to')).toEqual({
      name: 'designs',
      query: {
        version: undefined,
      },
    });
  });
});
