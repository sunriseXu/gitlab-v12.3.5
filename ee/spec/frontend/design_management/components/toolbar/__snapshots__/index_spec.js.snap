// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Design management toolbar component renders design and updated data 1`] = `
<header
  class="d-flex w-100 p-2 bg-white align-items-center js-design-header"
>
  <a
    aria-label="Go back to designs"
    class="mr-3 text-plain"
  >
    <icon-stub
      cssclasses=""
      name="close"
      size="18"
    />
  </a>
   
  <div>
    <h2
      class="m-0"
    >
      test.jpg
    </h2>
     
    <small
      class="text-secondary"
    >
      Updated 1 hour ago by Test Name
    </small>
  </div>
   
  <pagination-stub
    class="ml-auto"
    id="1"
  />
</header>
`;
